server/__tests__/userAccountAPI.test.js

```diff
@@ -0,0 +1,154 @@
+
+const chai = require('chai');
+const supertest = require('supertest');
+const app = require('../app/app');
+const { User } = require('../database/Models/User');
+const { getMockUser } = require('../utils/utils');
+
+const { expect } = chai;
+const { TEST_PORT } = process.env;
+
+let mockUser;
+let testServer;
+let request;
+
+beforeEach((done) => {
+  mockUser = getMockUser();
+  testServer = app.listen(TEST_PORT, () => {
+    request = supertest(testServer);
+    return done();
+  });
+});
+
+afterEach(done => testServer.close(done));
+
+// /api/signup test suite
+describe('Sign-Up API', () => {
+  it('registers a user with email, username and password', (done) => {
+    request.post('/api/signup')
+      .send(mockUser)
+      .expect(200)
+      .end((err, res) => {
+        const createdUser = res.body;
+        expect(createdUser).to.have.keys('_id', 'username');
+        return done(err);
+      });
+  });
+  it('give error codes when email or username or password are missing', (done) => {
+    const { email, username } = mockUser;
+    request.post('/api/signup')
+      .send({
+        email,
+        username,
+      })
+      .expect(422)
+      .end(() => done());
+  });
+  it('gives error messages when email or username or password are missing', (done) => {
+    const { email, username } = mockUser;
+    request.post('/api/signup')
+      .send({
+        email,
+        username,
+      })
+      .end((err, res) => {
+        expect(res.body).to.have.key('errors');
+        return done(err);
+      });
+  });
+  it('gives error message when username or email is already taken', (done) => {
+    // Create user first
+    request.post('/api/signup')
+      .send(mockUser)
+      .end(() => {
+        // Create same user
+        request.post('/api/signup')
+          .send(mockUser)
+          .expect(409)
+          .end((err, res) => {
+            expect(res.body).to.have.key('errors');
+            return done(err);
+          });
+      });
+  });
+  it('sends a JWT in the header on successful sign up', (done) => {
+    request.post('/api/signup')
+      .send(mockUser)
+      .end((error, response) => {
+        expect(response.header).to.include.key('x-auth-token');
+        return done(error);
+      });
+  });
+});
+
+// /api/login test suite
+describe('Login API', () => {
+  beforeEach((done) => {
+    // Create the user
+    User.create(mockUser, err => done(err));
```

Comment:
You also need to handle done on success. I guess? then the warning will go I think.

Reply:
@vkbinfo There is no separate handler for success. The callback is called everytime. If there is any error, done will resolve with that error, otherwise it will just resolve.